src/MizarWidgets/include/MizarWidgets/SamplingWithFrameTrackReportConfigValidator.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_REPORT_CONFIG_VALIDATOR_
+#define MIZAR_WIDGETS_SAMPLING_WITH_FRAME_TRACK_REPORT_CONFIG_VALIDATOR_
+
+#include <absl/functional/bind_front.h>
+
+#include <QMessageBox>
+#include <QString>
+
+#include "MizarBase/BaselineOrComparison.h"
+#include "MizarData/BaselineAndComparison.h"
+#include "MizarData/MizarPairedData.h"
+#include "MizarData/SamplingWithFrameTrackComparisonReport.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_mizar_widgets {
+
+template <typename BaselineAndComparison, typename PairedData, auto ReportError>
+class SamplingWithFrameTrackReportConfigValidatorTmpl {
+  template <typename T>
+  using Baseline = ::orbit_mizar_base::Baseline<T>;
+
+  template <typename T>
+  using Comparison = ::orbit_mizar_base::Comparison<T>;
+
+  using HalfConfig = orbit_mizar_data::HalfOfSamplingWithFrameTrackReportConfig;
+
+ public:
+  explicit SamplingWithFrameTrackReportConfigValidatorTmpl(
+      const Baseline<QString>& baseline_title, const Comparison<QString>& comparison_title)
+      : baseline_title_(baseline_title), comparison_title_(comparison_title) {}
+
+  [[nodiscard]] bool Validate(const BaselineAndComparison* baseline_and_comparison,
+                              const Baseline<HalfConfig>& baseline_config,
+                              const Comparison<HalfConfig>& comparison_config) const {
+    auto validator =
+        absl::bind_front(&SamplingWithFrameTrackReportConfigValidatorTmpl::ValidateAndReport, this);
+
+    Baseline<bool> baseline_ok = LiftAndApply(
```

Comment:
you are "dereferencing" the bool at the end anyways. I think it would be much more readable like this:

```
bool baseline_ok = ValidateAndReport(*baseline_config, *baseline_and_comparison->GetComparisonData(), *comparison_title_);
```

Reply:
Your suggestion uses two instances related to baseline and two instances related to comparison. The mistake is exactly the reason why I am doing this the way I do.

Reply:
Fair. But still the bools will need to be compared anyways. And for my excuse, githubs ui does not help me a lot when writting this suggestion.

Reply:
> But still the bools will need to be compared anyways.
I've considered adding a function like `And(Baseline<bool>, Comparison<bool>)` but this would be an overkill. In the worst case I would have written smth like `*baseline_ok && *baseline_ok` and I would be notified on an unused variable. 

The point of `Typedef`s is to stay in `Typedef`s as long as possible.

Reply:
> `Lift(validate_function)(config, data, title)`
I was thinking about smth like that. I am not sure if that could be as powerful as `LiftAndApply`. The latter has the full information about the types passed, while `Lift` would know only the type of the function. @beckerhe WDYT?